### ISSUE TITLE
mask rcnn on own data

### ISSUE THREAD

**Issue by lezardvt** [tensorflow/tpu#368]
Hi

I have follow the google tutorial to train Mask rcnn on Coco data without issue

I then followed the tutorial steps again but on my own data this time.

My dataset has about 3000 samples

This is how I start the train script:

python ~/tpu/models/official/mask_rcnn/mask_rcnn_main.py \
--use_tpu=True \
--tpu="tputputpu" \
--model_dir= "gs://my/path/mask-rcnn-model" \
--config="damconfig.yaml" \
--mode="train"

This is my config

num_classes: 9
backbone: 'resnet50'
use_bfloat16: True
train_batch_size: 16
eval_batch_size: 8
training_file_pattern: gs://my/path/TFRecords/train-*
validation_file_pattern: gs://my/path/TFRecords/val-*
val_json_file: gs://my/path/val_annotations.json
total_steps: 3000
num_steps_per_eval: 150
eval_samples: 311

I get the following error when I start training:

INFO:tensorflow:Enqueue next (2500) batch(es) of data to infeed.
INFO:tensorflow:Dequeue next (2500) batch(es) of data from outfeed.
INFO:tensorflow:Error recorded from infeed: assertion failed: [103]
         [[{{node parser/Assert_2/Assert}}]]
         [[node input_pipeline_task0/while/IteratorGetNext_4 (defined at /usr/local/lib/python2.7/dist-packages/tensorflow_estimator/python/estimator/estimator.py:1112) ]]
INFO:tensorflow:An error was raised. This may be due to a preemption in a connected worker or parameter server. The current session will be closed and a new session will be created. This error may also occur due to a gRPC failure caused by high memory or network bandwidth usage in the parameter servers. If this error occurs repeatedly, try increasing the number of parameter servers assigned to the job. Error: Socket closed

Any help or suggestions would be much appreciated

**Comment by saberkun** [comment]
This may be due to a preemption in a connected worker or parameter server.  Are you using preemptible TPU? Maybe this is transient error.

**Comment by lezardvt** [comment]
I have tried non-preemptible V2-8 and V3-8 TPUs and I still get the error. 

I found that when I reduce my dataset to ~700 images the error goes away and it trains nicely. 

Could it be this part of the error message then? - "This error may also occur due to a gRPC failure caused by high memory or network bandwidth usage in the parameter servers"

**Comment by saberkun** [comment]
https://github.com/tensorflow/tpu/blob/master/models/official/mask_rcnn/dataloader.py#L301
Something around here might be related. You may need to tune down the tf.data buffer or less parallelism.

**Comment by lezardvt** [comment]
> https://github.com/tensorflow/tpu/blob/master/models/official/mask_rcnn/dataloader.py#L301
> Something around here might be related. You may need to tune down the tf.data buffer or less parallelism.

Thanks for the help. 

Solved the problem. In the file that you referenced the are 2 variables : 
MAX_NUM_INSTANCES = 100
MAX_NUM_POLYGON_LIST_LEN = 2600

I had to fine tune these to fit my training data. My dataset has many polygons per image.
